### PR TITLE
Revise delete tests

### PR DESCRIPTION
**Pull request by benjaminjb**
**Checklist:**
 - [ ] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change, as applicable?
 - [ ] Have you tested your changes on all related environments with successful results, as applicable?
   - [ ] Have you added automated tests?

**Type of Changes:**
 - [ ] New feature
 - [ ] Bug fix
 - [ ] Documentation
 - [x] Testing enhancement
 - [ ] Other

**What is the current behavior (link to any open issues here)?**

**What is the new behavior (if this is a feature change)?**
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

* Use files for legibility
* Add describe/log collectors to every assert

**Other Information**:

### CODE REVIEW

File: testing/kuttl/e2e/delete/00--create-cluster.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestStep
+apply:
+- files/00-create-namespace.yaml
```
Comment (jmckulk) [inline testing/kuttl/e2e/delete/00--create-cluster.yaml:4]: I don't see a `files/00-create-namespace.yaml`

File: testing/kuttl/e2e/delete/10--create-cluster.yaml
Comment (jmckulk) [inline testing/kuttl/e2e/delete/10--create-cluster.yaml:1]: Maybe this test could be in a different directory?
```
testing/kuttl/e2e
  delete/
    00-create...
  delete-<descriptive-suffix>/
    10-create...
```

also maybe not, having each delete type in a directory makes sense too.

A better name on `10-*` would be appreciated. To see that it is deleting a replica cluster without having to click into the create-cluster.yaml file or the readme

File: testing/kuttl/e2e/delete/20--create-cluster.yaml
Comment (jmckulk) [inline testing/kuttl/e2e/delete/20--create-cluster.yaml:1]: same for this file name

File: testing/kuttl/e2e/delete/README.md [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 
-#### Delete cluster with replica
+#### Delete cluster with replica (10-11)
 
 * Start a regular cluster with 2 replicas
 * Delete it
 * Check that nothing remains
 
-#### Delete a cluster that never started
+#### Delete a cluster that never started (20-21)
```
Comment (jmckulk) [inline testing/kuttl/e2e/delete/README.md:15]: Do we need to check this? Kube should cleanup the resources regardless of the cluster status

File: testing/kuttl/e2e/delete/01-errors.yaml
Comment (jmckulk) [inline testing/kuttl/e2e/delete/01-errors.yaml:1]: would it make sense to put this in a TestStep - to be consistent with the delete-namespace test/other tests?

**Review by jmckulk — APPROVED** [review]
lgtm

File: testing/kuttl/e2e/delete-namespace/00-assert.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestAssert
+collectors:
+- type: command
+  command: kubectl -n ${KUTTL_TEST_DELETE_NAMESPACE} describe pods --selector postgres-operator.crunchydata.com/cluster=delete-namespace
+- namespace: ${KUTTL_TEST_DELETE_NAMESPACE}
+  selector: postgres-operator.crunchydata.com/cluster=delete-namespace
```
Comment (tjmoore4) [inline testing/kuttl/e2e/delete-namespace/00-assert.yaml:7]: :pick: Missing newlines on a few of the files.

**Review by tjmoore4 — APPROVED** [review]
LGTM

File: testing/kuttl/e2e/delete/00--create-cluster.yaml (hunk 1 above)
Comment (benjaminjb) [reply testing/kuttl/e2e/delete/00--create-cluster.yaml:4]: copy-paste error

File: testing/kuttl/e2e/delete/10--create-cluster.yaml
Comment (benjaminjb) [reply testing/kuttl/e2e/delete/10--create-cluster.yaml:1]: Better names, check

File: testing/kuttl/e2e/delete/README.md (hunk 2 above)
Comment (benjaminjb) [reply testing/kuttl/e2e/delete/README.md:15]: Hmmm, I think this test was to check that we didn't have a finalizer on the cluster that made it impossible to delete while not running.

**Pull request merged**